### PR TITLE
Add @since tags to all methods in public namespace

### PR DESCRIPTION
**Pull request by MorrisJobke**
- enhance the app development experience - you can look up the
  method introduction right inside the code without searching
  via git blame
- easier to write apps for multiple versions
- I didn't add those labels for some already deprecated methods

cc @LukasReschke @Deepdiver1975 @PVince81 @Raydiation

@karlitschek I hope this helps a lot to get a better app ecosystem. Now devs can develop on recent versions and know how good it will work in older versions. :rocket:

### CODE REVIEW

**Comment by MorrisJobke** [comment]
Because the question came up: If the method signatur changed I added a comment in which version a parameter was added/a default value was set:

https://github.com/owncloud/core/pull/15680/files#diff-ce940861b2022699606a2c98128ab2bfR71

File: lib/public/appframework/controller.php [hunk 1]
```diff
@@ -34,32 +34,41 @@
 use OCP\AppFramework\Http\TemplateResponse;
 use OCP\AppFramework\Http\JSONResponse;
 use OCP\AppFramework\Http\DataResponse;
+use OCP\AppFramework\Http\Response;
```
Comment (MorrisJobke) [inline lib/public/appframework/controller.php]: This is the only actual code change in this whole diff - it was marked read in the PHPDoc so I added it as use statement to fix that.

File: lib/public/activity/iextension.php [hunk 2]
```diff
@@ -31,6 +31,12 @@
 // This means that they should be used by apps instead of the internal ownCloud classes
 namespace OCP\Activity;
 
+/**
+ * Interface IExtension
+ *
+ * @package OCP\Activity
+ * @since 8.0.0
```
Comment (DeepDiver1975) [inline lib/public/activity/iextension.php]: wasn't that 6.0 as well?

File: lib/public/app.php [hunk 3]
```diff
@@ -73,6 +74,7 @@ public static function register( $data ) {
 	 *
 	 * @deprecated Use \OC::$server->getNavigationManager()->add() instead to
 	 * register a closure, this helps to speed up all requests against ownCloud
+	 * @since 4.0.0
```
Comment (DeepDiver1975) [inline lib/public/app.php]: would be great to have an attribute like @deprecated-since 6.0.0

File: lib/public/activity/iextension.php (hunk 2 above)
Comment (MorrisJobke) [inline lib/public/activity/iextension.php]: No: 8363f3c6355e5bd2bb54875d11e326747df265c8

File: lib/public/app.php (hunk 3 above)
Comment (MorrisJobke) [inline lib/public/app.php]: @DeepDiver1975 I will cleanup the @deprecated stuff afterwards in a separate PR.

**Comment by rullzer** [comment]
Very cool!
:+1:, I obviously did not check the stuff in detail since I have no desire to duplicate your manual work.

**Comment by DeepDiver1975** [comment]
I like that - THX @MorrisJobke

**Comment by DeepDiver1975** [comment]
:+1:

**Pull request merged**